src/hex_tarball.erl

```diff
@@ -0,0 +1,455 @@
+-module(hex_tarball).
```

Comment:
some other ideas for name are:

- `hex`
- `hex_package`
- `hex_erl_{tarball|package}`

I kinda like `hex_package` the most because then `hex_package:tarball()` type looks better than `hex_tarball:tarball()` :-)

other ideas?

PS I briefly considered having type `t() :: #{checksum := ..., contentes := ..., metadata := ...}` which would be really nice because it exactly matches the spec, but this type would have been used only when unpacking to memory (when unpacking to disk we don't return contents)

PS2 I wonder if there's gonna be `hex_erl` module at all. I think we could add later `hex_erl:version/1` so that when we vendor this it could be used to generate stuff like:

```
cat vendored_hex_tarball.erl
%% Vendored from hex_erl v0.1.0-dev, do not edit manually
```

Reply:
I think I prefer the current name since the module is specifically about the package tarball and we can have other modules like the package registry etc.. I was thinking `hex_erl` for the project/application name and use the `hex_` namespace.

Reply:
👍